source/extensions/geoip_providers/maxmind/geoip_provider.cc

```diff
@@ -88,29 +103,88 @@ GeoipProvider::~GeoipProvider() {
   ENVOY_LOG(debug, "Shutting down Maxmind geolocation provider");
   if (city_db_) {
     MMDB_close(city_db_.get());
+    city_db_ = nullptr;
```

Comment:
why is this change needed? Database pointer is a unique smart pointer owned by provider instance, memory will be released by smart pointer on geoip provider destruction.